icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/RbnfTest.java

```diff
@@ -1207,6 +1255,32 @@ else if (testParsing) {
         }
     }
 
+    void doLenientTest(RuleBasedNumberFormat formatter, String[][] testData) {
+        NumberFormat decFmt = new DecimalFormat("#,###.################");
+        formatter.setLenientParseMode(true);
```

Comment:
Please refactor this to not have copypasta (redundant code). It makes it harder to maintain the code when the logic needs to change. You can create a new one function that just defaults the lenient mode argument.

Reply:
@grhoten,
Hmm... I didn't get. In this case, the method **doLenientTest()** is different from **doTest()**. 

_setLenientParseMode(true)_ is called only in **doLenientTest()**.

For _NumberFormat decFmt = new DecimalFormat("#,###.################");_, I agree it was copy/paste, so it makes sense to refactor to a method.

Reply:
Typically people use default arguments in such scenarios, like refactoring it the following way:

```
public void doTest(args...)
{
    doTest(args, false);
}
public void doTest(args..., boolean lenient)
{
...
    if (lenient) {
        rbnf.setLenient(true);
    }
...
}
```

Reply:
Got it, got it. So we can even reuse the doTest stuff in future, in case we want.

That's fine, George. I'll do an ammend.